crates/orchestrator/src/jobs/state_update_job/mod.rs

```diff
@@ -31,26 +53,207 @@ impl Job for StateUpdateJob {
         })
     }
 
-    async fn process_job(&self, _config: &Config, _job: &JobItem) -> Result<String> {
+    async fn process_job(&self, config: &Config, job: &mut JobItem) -> Result<String> {
+        let attempt_no =
+            job.metadata.get(JOB_PROCESS_ATTEMPT_METADATA_KEY).expect("Could not find current attempt number.").clone();
+
+        // TODO: remove when SNOS is correctly stored in DB/S3
+        // Test metadata to fetch the snos output from the test folder, default to False
+        let fetch_from_tests =
+            job.metadata.get(JOB_METADATA_STATE_UPDATE_FETCH_FROM_TESTS).map_or(false, |value| value == "TRUE");
+
         // Read the metadata to get the blocks for which state update will be performed.
-        // For each block, get the program output (from the PIE?) and the
-        todo!()
+        // We assume that blocks nbrs are formatted as follow: "2,3,4,5,6".
+        let block_numbers = self.get_block_numbers_from_metadata(job)?;
+        self.validate_block_numbers(config, &block_numbers).await?;
+
+        let mut sent_tx_hashes: Vec<String> = Vec::with_capacity(block_numbers.len());
+        for block_no in block_numbers.iter() {
+            let snos = self.fetch_snos_for_block(*block_no, Some(fetch_from_tests)).await;
+            let tx_hash = self.update_state_for_block(config, *block_no, snos, Some(fetch_from_tests)).await?;
+            sent_tx_hashes.push(tx_hash);
+        }
+
+        // Insert the tx hashes into the the metadata for the attempt number - will be used later by
+        // verify_job to make sure that all tx are successful.
+        let new_attempt_metadata_key = format!("{}{}", JOB_METADATA_STATE_UPDATE_ATTEMPT_PREFIX, attempt_no);
```

Comment:
let's make the prefix `attempt_tx_hashes_{i}`?

Reply:
Done